### PR TITLE
add desktop type for rhel, centos, windows os

### PR DESCRIPTION
**Pull request by ksimon1**
This PR adds desktop type for rhel, centos, windows os.
Desktop type templates (rhel, centos) and fedora, ubuntu will contains table devices

### CODE REVIEW

File: automation/test.sh [hunk 1]
```diff
@@ -238,63 +238,66 @@ _oc exec -it winrmcli -- yum install -y iproute iputils
...
-  windowsTemplatePath="../../dist/templates/win2k12r2-generic-$size.yaml"
-
-  _oc process -o json --local -f $windowsTemplatePath NAME=win2k1r2-$size PVCNAME=disk-windows | \
-  jq '.items[0].spec.template.spec.volumes[0]+= {"ephemeral": {"persistentVolumeClaim": {"claimName": "disk-windows"}}} | 
-  del(.items[0].spec.template.spec.volumes[0].persistentVolumeClaim)' | \
-  _oc apply -f -
-
-  # start vm
-  ./virtctl --kubeconfig=$kubeconfig start win2k1r2-$size
+  for types in ${types[@]}; do
```
Comment (MarSik) [inline automation/test.sh]: types -> type?

File: automation/test.sh [hunk 2]
```diff
@@ -238,63 +238,66 @@ _oc exec -it winrmcli -- yum install -y iproute iputils
...
+    windowsTemplatePath="../../dist/templates/win2k12r2-generic$type-$size.yaml"
```
Comment (MarSik) [inline automation/test.sh]: Shouldn't the generic string be removed?

File: automation/test.sh [hunk 3]
```diff
@@ -238,63 +238,66 @@ _oc exec -it winrmcli -- yum install -y iproute iputils
...
+
+    _oc process -o json --local -f $windowsTemplatePath NAME=win2k1r2-$size PVCNAME=disk-windows | \
+    jq '.items[0].spec.template.spec.volumes[0]+= {"ephemeral": {"persistentVolumeClaim": {"claimName": "disk-windows"}}} | 
+    del(.items[0].spec.template.spec.volumes[0].persistentVolumeClaim)' | \
+    _oc apply -f -
+
+    # start vm
+    ./virtctl --kubeconfig=$kubeconfig start win2k1r2-$size
```
Comment (MarSik) [inline automation/test.sh]: -$type-$size?

File: automation/test.sh [hunk 4]
```diff
@@ -238,63 +238,66 @@ _oc exec -it winrmcli -- yum install -y iproute iputils
...
+
+    set +e
+    current_time=0
+    while [[ $(_oc get vmi win2k1r2-$size -o json | jq '.status.phase') != *Running* ]] ; do 
+      _oc describe vmi win2k1r2-$size
```
Comment (MarSik) [inline automation/test.sh]: Same here, shouldn't type be part of the name?

File: automation/test.sh [hunk 5]
```diff
@@ -238,63 +238,66 @@ _oc exec -it winrmcli -- yum install -y iproute iputils
...
+        exit 1
+      fi
+      sleep $sample;
+    done
+    set -e
 
-  set +e
-  current_time=0
-  while [[ $(_oc get vmi win2k1r2-$size -o json | jq '.status.phase') != *Running* ]] ; do 
+    _oc describe vm win2k1r2-$size
```
Comment (MarSik) [inline automation/test.sh]: same..

File: automation/test.sh [hunk 6]
```diff
@@ -238,63 +238,66 @@ _oc exec -it winrmcli -- yum install -y iproute iputils
...
-  current_time=0
-  # Make sure vm is ready
-  while _oc exec -it winrmcli -- ping -c1 $ipAddressVMI| grep "Destination Host Unreachable" ; do 
-    current_time=$((current_time + 10))
-    if [ $current_time -gt $timeout ]; then
-      exit 1
-    fi
-    sleep 10;
-  done
+    ./virtctl --kubeconfig=$kubeconfig stop win2k1r2-$size
```
Comment (MarSik) [inline automation/test.sh]: same...

File: automation/test.sh [hunk 7]
```diff
@@ -238,63 +238,66 @@ _oc exec -it winrmcli -- yum install -y iproute iputils
...
-  timeout=300
-  current_time=0
-  # run ipconfig /all command on windows vm
-  while [[ $(_oc exec -it winrmcli -- ./usr/bin/winrm-cli -hostname $ipAddressVMI -port 5985 -username "Administrator" -password "Heslo123" "ipconfig /all") != *"$ipAddressVMI"* ]] ; do 
-    current_time=$((current_time + 10))
-    if [ $current_time -gt $timeout ]; then
-      exit 1
-    fi
-    sleep 10;
+    _oc process -o json --local -f $windowsTemplatePath NAME=win2k1r2-$size PVCNAME=disk-windows | \
```
Comment (MarSik) [inline automation/test.sh]: same...

File: generate-templates.yaml [hunk 8]
```diff
@@ -35,6 +35,33 @@
        - rhel7.6
       osinfoname: "{{ oslabels[0] }}"
 
+  - name: Generate desktop RHEL 7 templates
+    template:
+      src: rhel7.tpl.yaml
+      dest: "{{ playbook_dir }}/dist/templates/{{ os }}-{{ item.workload }}-{{ item.type }}-{{ item.flavor }}.yaml"
+    with_items:
+    - {type: desktop, flavor: tiny, workload: generic, memsize: "1G", cpus: 1, iothreads: False}
```
Comment (MarSik) [inline generate-templates.yaml]: Actually the workload generic is supposed to be split into desktop, server and generic (= server)

Comment (MarSik) [reply generate-templates.yaml]: There is no need to invent `type` as well.

File: automation/test.sh (hunk 1 above)
Comment (MarSik) [reply automation/test.sh]: And actually, the type here should be called workload

File: .travis.yml [hunk 9]
```diff
@@ -20,6 +20,10 @@ jobs:
     env: TARGET=functional-tests TEST_FUNCTIONAL="fedora-generic-tiny fedora-generic-small fedora-generic-medium fedora-generic-large"
     script: bash -x test.sh
 
+  - name: Functional test of the desktop Fedora template
+    env: TARGET=functional-tests TEST_FUNCTIONAL="fedora-generic-desktop-tiny fedora-generic-desktop-small fedora-generic-desktop-medium fedora-generic-desktop-large"
```
Comment (MarSik) [inline .travis.yml]: fedora-desktop-small, the desktop part is a workload specification, we are not adding any new type property

**Comment by MarSik** [comment]
I think there was a slight misunderstanding about what type desktop means. The idea was to add additional workloads desktop and server. Generic should stay equivalent to server.

**Comment by ksimon1** [comment]
@MarSik please review

**Pull request merged**